packages/myst-cli/src/process/notebook.ts

```diff
@@ -70,13 +74,27 @@ export async function processNotebook(
   file: string,
   content: string,
 ): Promise<GenericParent> {
+  const { mdast } = await processNotebookFull(session, file, content);
+  return mdast;
+}
+
+export async function processNotebookFull(
+  session: ISession,
+  file: string,
+  content: string,
+): Promise<{ mdast: GenericParent; frontmatter: PageFrontmatter }> {
   const { log } = session;
   const { metadata, cells } = JSON.parse(content) as INotebookContent;
   // notebook will be empty, use generateNotebookChildren, generateNotebookOrder here if we want to populate those
 
   const language = metadata?.kernelspec?.language ?? 'python';
   log.debug(`Processing Notebook: "${file}"`);
 
+  // Load frontmatter from notebook metadata
+  const vfile = new VFile();
+  vfile.path = file;
+  const frontmatter = validatePageFrontmatter(metadata ?? {}, frontmatterValidationOpts(vfile));
```

Comment:
@agoose77 what exactly ends up in this object here? is it all notebook metadata or will the frontmatter be pruned down to known fields only?

Reply:
just checking through the validate function I *think* that is what will happen, i.e. weare only allowing known fields into the processed notebook json, rather than the whole metadata...

Reply:
Yes, the `frontmatter` result should be the validated object - we should be dropping unknown fields.